plugins/coordinator/plugin.go

```diff
@@ -0,0 +1,153 @@
+package coordinator
+
+import (
+	"os"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+
+	"github.com/iotaledger/hive.go/daemon"
+	"github.com/iotaledger/hive.go/logger"
+	"github.com/iotaledger/hive.go/node"
+	"github.com/iotaledger/hive.go/syncutils"
+	"github.com/iotaledger/hive.go/timeutil"
+	"github.com/iotaledger/iota.go/consts"
+	"github.com/iotaledger/iota.go/guards"
+	"github.com/iotaledger/iota.go/pow"
+	"github.com/iotaledger/iota.go/trinary"
+
+	"github.com/gohornet/hornet/pkg/config"
+	"github.com/gohornet/hornet/pkg/model/milestone"
+	"github.com/gohornet/hornet/pkg/shutdown"
+	tanglePlugin "github.com/gohornet/hornet/plugins/tangle"
+)
+
+func init() {
+	pflag.CommandLine.MarkHidden("cooBootstrap")
+	pflag.CommandLine.MarkHidden("cooStartIndex")
+}
+
+var (
+	PLUGIN = node.NewPlugin("Coordinator", node.Disabled, configure, run)
+	log    *logger.Logger
+
+	// config options
+	seed                   trinary.Hash
+	securityLvl            int
+	merkleTreeDepth        int
+	mwm                    int
+	stateFilePath          string
+	merkleTreeFilePath     string
+	milestoneIntervalSec   int
+	checkpointTransactions int
+
+	coordinatorState      *CoordinatorState
+	coordinatorMerkleTree *MerkleTree
+	milestoneLock         = syncutils.Mutex{}
+	lastCheckpointCount   int
+	lastCheckpointHash    *trinary.Hash
+	_, powFunc            = pow.GetFastestProofOfWorkImpl()
+	bootstrapped          = false
+	bootstrap             = pflag.Bool("cooBootstrap", false, "bootstrap the network")
+	startIndex            = pflag.Uint32("cooStartIndex", 0, "first index of network")
```

Comment:
Include this parameters within the Coordinator struct instance  instead of having them declared like this.